python/lvmdrp/functions/run_drp.py

```diff
@@ -673,17 +685,20 @@ def run_drp(mjd: Union[int, str, list], bias: bool = False, dark: bool = False,
         sub = filter_expnum(sub, expnum)
 
     # get biases and darks
-    cond = sub['imagetyp'].isin(['bias', 'dark'])
+    cond = sub['imagetyp'].isin(['bias', 'dark', 'pixelflat'])
     precals = sub[cond]
     if len(precals) == 0 and not skip_bd:
         log.error(f'No biases or darks found for mjd {mjd}. Discontinuing reduction.')
         return
     precals = precals.sort_values(['expnum', 'camera'])
 
     if not skip_bd:
+        # when to create pixel mask
+        on_pixflats = 'pixelflat' in set(precals['imagetyp'])
         # reduce biases / darks
-        reduce_set(precals, settype="precals", flavor='bias')
-        reduce_set(precals, settype="precals", flavor='dark')
+        reduce_set(precals, settype='precals', flavor='bias')
+        reduce_set(precals, settype='precals', flavor='dark', create_pixmask=not on_pixflats)
+        reduce_set(precals, settype='precals', flavor='pixelflat', create_pixmask=on_pixflats)
```

Comment:
Since `create_pixmask` defaults to False, we only need to set this when it's True.  Technically I don't think we really need the `create_pixmask` keyword, if you move the conditional inside `reduce_set`, but it's ok to leave it. 

I would suggest
```
reduce_set(precals, settype="precals", flavor='bias')
reduce_set(precals, settype="precals", flavor='dark')
reduce_set(precals, settype="precals", flavor='pixelflat', create_pixmask=on_pixflats)
```